### PR TITLE
use POST email address methods

### PR DESCRIPTION
**Pull request by gidsg**
This updates the API client to use the methods introduced in https://github.com/alphagov/digitalmarketplace-api/pull/1037

### CODE REVIEW

**Comment by lfdebrux** [comment]
### My review checklist

I've:

- [ ] looked at the code
- [ ] tested locally
- [ ] run functional tests locally

File: tests/test_data_api_client.py [hunk 1]
```diff
@@ -632,8 +632,8 @@ def test_email_address_with_valid_admin_domain_is_true(self, data_client, rmock)
         assert result is True
 
     def test_email_address_with_invalid_admin_domain_is_false(self, data_client, rmock):
-        rmock.get(
-            "http://baseurl/users/valid-admin-email?email_address=kev%40not-gov.uk",
+        rmock.post(
+            "http://baseurl/users/valid-admin-email",
             json={"valid": False},
             status_code=200
         )
```
Comment (katstevens) [inline tests/test_data_api_client.py:639]: For POST requests, we can also assert the JSON from the last call (see line 558 for an example):

```
assert rmock.last_request.json() == {
    "emailAddress": True
}
```

Comment (gidsg) [reply tests/test_data_api_client.py:640]: great suggestion @katstevens have added these assertions in 9b3ebff.

Comment (katstevens) [reply tests/test_data_api_client.py:640]: Great 👍 (could we add for the `test_is_email_address_with_valid_buyer_domain_(true|false)` tests as well?)

Comment (gidsg) [reply tests/test_data_api_client.py:642]: good spot, missed those. Added those now in 55c6c37.

**Comment by lfdebrux** [comment]
Functional tests passed for me.

**Pull request merged**